pkg/recipes/driver/terraform.go

```diff
@@ -152,14 +152,17 @@ func (d *terraformDriver) Delete(ctx context.Context, opts DeleteOptions) error
 // prepareRecipeResponse populates the recipe response from the module output named "result" and the
 // resources deployed by the Terraform module. The outputs and resources are retrieved from the input Terraform JSON state.
 func (d *terraformDriver) prepareRecipeResponse(ctx context.Context, definition recipes.EnvironmentDefinition, tfState *tfjson.State) (*recipes.RecipeOutput, error) {
-	if tfState == nil || (*tfState == tfjson.State{}) {
+	// We need to use reflect.DeepEqual to compare the struct that has a slice with an empty struct.
+	// The reason is that Go does not allow comparison of structs that contain slices.
+	// Please see: https://go.dev/ref/spec#Comparison_operators.
+	if tfState == nil || reflect.DeepEqual(*tfState, tfjson.State{}) {
 		return &recipes.RecipeOutput{}, errors.New("terraform state is empty")
 	}
 
 	recipeResponse := &recipes.RecipeOutput{}
-	moduleOutputs := tfState.Values.Outputs
-	if moduleOutputs != nil {
+	if tfState.Values != nil && tfState.Values.Outputs != nil {
```

Comment:
Could you look at this part and let me know if this makes sense @kachawla @vishwahiremat? Can the **Values** ever be nil?

Reply:
I am not sure if values can be Nil, but i think its good to have the nil check on Values.

Reply:
It is possible for extender resource since we don't do enforce any outputs for that, so I think it makes sense to add the nil check. Thanks @ytimocin!